src/util/manifest.js

```diff
@@ -85,6 +86,20 @@ export default async function getValidatedManifest(
 
 
 export function getManifestId(manifestData: ExtensionManifest): string | void {
-  return manifestData.applications ?
-    manifestData.applications.gecko.id : undefined;
+  if (
+    manifestData.applications &&
+    manifestData.applications.gecko.id
```

Comment:
it seems reasonable to also check that the `applications.gecko` property actually exists here, as you are doing for the `browser_specific_settings.gecko` one below.

Reply:
I initially added this but then realised the existing code didn't have this check [look like for this reason](https://github.com/mozilla/web-ext/blob/995ccf5b8e651cdfec336be257e908d9c2ee4a03/src/util/manifest.js#L72)

Reply:
Updated the PR + types to cover this use case